usaspending_api/common/filters/time_period/decorators/new_awards_only.py

```diff
@@ -0,0 +1,75 @@
+from usaspending_api.common.filters.time_period.time_period import ITimePeriod
+from usaspending_api.search.filters.elasticsearch.filter import _QueryType
+
+NEW_AWARDS_ONLY_KEYWORD = "new_awards_only"
+
+
+class NewAwardsOnlyTimePeriod(ITimePeriod):
+    """A decorator class that can be used to apply a new awards only filter
+    ON TOP of an existing time period filter.
+    """
+
+    def __init__(self, transaction_search_time_period_obj, query_type, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._query_type = query_type
+        self._transaction_search_time_period_obj = transaction_search_time_period_obj
+        self._additional_gte_filters_for_new_awards_only = {
+            _QueryType.TRANSACTIONS: [
+                {"action_date": {"gte": self.start_date()}},
```

Comment:
> May be best in this case just to duplicate the logic rather than trying to get maximal reuse. e.g.

Duplicating logic is the whole reason why these bugs exist.

Reply:
Yep, I saw the change with taking this out of the constructor to capture them in a implementation-specific property. That'll work.